package/MDAnalysis/analysis/base.py

```diff
@@ -238,17 +239,29 @@ def _setup_frames(self, trajectory, start=None, stop=None, step=None):
             stop frame of analysis
         step : int, optional
             number of frames to skip between each analysed frame
+        frames : array_like, optional
+            array of integers or booleans to slice trajectory
 
 
         .. versionchanged:: 1.0.0
             Added .frames and .times arrays as attributes
+
+        .. versionchanged:: 2.1.0
+            Added frames processing
         """
         self._trajectory = trajectory
-        start, stop, step = trajectory.check_slice_indices(start, stop, step)
+        if frames is not None:
+            if any([start, stop, step]):
+                raise ValueError("start/stop/step cannot be combined with frames")
+            slicer = frames
+        else:
+            start, stop, step = trajectory.check_slice_indices(start, stop, step)
+            slicer = slice(start, stop, step)
+        self._sliced_trajectory = trajectory[slicer]
```

Comment:
I _think_ using a frameiterator changes the behaviour of the iteration.

So if I do:

```python
import MDAnalysis as mda
from MDAnalysisTests.datafiles import TPR, XTC

u = mda.Universe(TPR, XTC)

iterator = u.trajectory[[0,1,2]]

for ts in iterator:
    pass

print(u.trajectory.ts)
```
Ouput: < Timestep 2 with unit cell dimensions [79.98359 79.98359 79.98358 60.      60.      90.     ] >

Whilst:

```python
import MDAnalysis as mda
from MDAnalysisTests.datafiles import TPR, XTC

u = mda.Universe(TPR, XTC)

for ts in u.trajectory[:3]
    pass

print(u.trajectory.ts)
```
Ouput: < Timestep 0 with unit cell dimensions [80.017006 80.017006 80.017006 60.       60.       90.      ] >

This is a general thing with the way we iterate through a trajectory based on an index selection (not limited to this code here), and we should probably fix this? But I'd say it's probably a bad idea for the two approaches differ like this for AnalysisBase.

Reply:
I've opened a separate issue, I'm not actually sure if it's intended behaviour.

Reply:
Thanks for opening  #3416.

As a band-aid solution we could add a `u.trajectory.rewind()` after the `for i, ts in enumerate(self._sliced_trajectory)` loop. This would ensure consistent behavior of all AnalysisBase classes. It would just need a comment that references issue #3416 so that we can remove it again if that gets fixed/changed.

Reply:
Sure I'm fine with this as a temporary measure. We should implement the fix in this PR so that we don't forget.

Reply:
I added the line to rewind with a comment referencing the issue